hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipUpdateTest.java

```diff
@@ -799,18 +798,23 @@ public void testListenerRegistrationWhileANewMemberJoining() {
         CountDownLatch latch = new CountDownLatch(1);
         Config config = getConfigWithService(new PostJoinAwareServiceImpl(latch), PostJoinAwareServiceImpl.SERVICE_NAME);
 
-        EntryListenerConfig listenerConfig = new EntryListenerConfig();
         CountDownLatch entryAddedLatch = new CountDownLatch(2);
-        listenerConfig.setImplementation((EntryAddedListener) event -> entryAddedLatch.countDown());
-        config.getMapConfig("test").addEntryListenerConfig(listenerConfig);
-
         HazelcastInstance hz1 = factory.newHazelcastInstance(config);
+
+        // register a listener on the first member
+        hz1.getMap("test").addEntryListener((EntryAddedListener) event -> entryAddedLatch.countDown(), true);
```

Comment:
Thx for reminding this.

>I assume that if we register a listener via Config, we couldn't confidently test that it will be propagated to a new member because the same Config is passed to both members.

Listener should receive all events from the cluster unless it is not marked as a local listener. There must be no difference whether or not you registered it via config object. 


>The test logic also was changed based on Vassilis's response:
A: Do we guarantee that “with 0 backups, a member-side EntryListener with local == false will receive an event for an entry that was added on any member of the cluster”?
V: no, our event system does not guarantee that. It can drop events under load

Even this is a legitimate possibility, I think it is a rare case and there is no load to cause it in the failing test.

Maybe core issue could be something else caused during joins or something unknown now, i think.

Reply:
Thanks, @ahmetmircik for the comments!
I've re-analysed the issues and updated the test-fix.